ng-test/src/app/address.geo/address-geo.component.ts

```diff
@@ -0,0 +1,65 @@
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { LoaderService, UserAddress, ShopAddress } from '../../../../dist';
+import { HttpClient } from '@angular/common/http';
+
+@Component({
+  selector: 'app-address-geo',
+  templateUrl: './address-geo.component.html',
+  styleUrls: ['./address-geo.component.scss']
+})
+export class AddressGeoComponent implements OnInit {
+
+  constructor(
+    private http: HttpClient
+  ) { }
+
+  ngOnInit() {
+    this.addressReady = false;
+  }
+
+  @Input() addressUser: UserAddress;
+  @Input() addressShop: ShopAddress;
+  @Input() addressReady :boolean;
+  @Output() addressReadyEvent = new EventEmitter<any>();
```

Comment:
excellent Matthieu! 
pourquoi tu différencies `UserAddress` et `ShopAddress`? non il faut que l'address soit indépendant de l'usage quitte a ajouter un discriminant dans le constructeur. 

tu peux enlver `addressReady`

je te propose de renommer ton feedback  `onAddressReady(address||false)`  pour avoir les deux cas,
- OK: `onAddressReady(this.address)`
- KO:`onAddressReady(false)`

Reply:
Pourquoi retourner l'address avec l'event ? elle est déjà bind (2way) depuis le composant parent donc pas besoin de la retourner ou je me trompe ?

Reply:
Et pourquoi enlever addressReady ? je l'utilise pour afficher ou non la map. 
```
<div class="row" *ngIf="addressReady">
  <img src="//maps.googleapis.com/maps/api/staticmap?center={{address?.geo?.lat}},{{address?.geo?.lng}}&zoom=12&size=400x400&markers=color:green%7Xlabel:%7C{{address.geo.lat}},{{address.geo.lng}}&key=AIzaSyD5w46BmWX6hX-uJ2yMycS_cRb2HRvDXQU"/>
</div>
```

Reply:
Okay j'ai enlevé ``` addressReady ```